benches/setup.rs

```diff
@@ -0,0 +1,110 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::thread::LocalKey;
+
+use dapol::accumulators::{NdmSmt, NdmSmtConfigBuilder};
+use dapol::{read_write_utils, EntityId, Height, InclusionProof, MaxThreadCount};
+use log::info;
+
+// CONSTANTS
+// ================================================================================================
+
+pub const TREE_HEIGHTS: [u8; 3] = [16, 32, 64];
+pub const NUM_USERS: [u64; 35] = [
+    10_000,
+    20_000,
+    30_000,
+    40_000,
+    50_000,
+    60_000,
+    70_000,
+    80_000,
+    90_000,
+    100_000,
+    200_000,
+    300_000,
+    400_000,
+    500_000,
+    600_000,
+    700_000,
+    800_000,
+    900_000,
+    1_000_000,
+    2_000_000,
+    3_000_000,
+    4_000_000,
+    5_000_000,
+    6_000_000,
+    7_000_000,
+    8_000_000,
+    9_000_000,
+    10_000_000,
+    30_000_000,
+    50_000_000,
+    70_000_000,
+    90_000_000,
+    100_000_000,
+    125_000_000,
+    250_000_000,
+];
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+pub fn build_ndm_smt(tup: (Height, MaxThreadCount, u64)) -> NdmSmt {
+    let height_int = tup.0.as_raw_int();
+    let max_users_for_height = 2_u64.pow((height_int - 1) as u32);
+
+    if tup.2 > max_users_for_height {
+        panic!("Number of users exceeds max for height {:?}", height_int);
+    }
+
+    if tup.1.get_value() > MaxThreadCount::default().get_value() {
+        panic!("Max thread count exceeds available machine parallelism");
```

Comment:
If a bench panics then does the whole process die or does it move onto the next bench?

Reply:
I changed the `panic` to an `Err`